src/backend/executor/spi.c

```diff
@@ -1430,19 +1429,6 @@ SPI_is_cursor_plan(SPIPlanPtr plan)
 	}
 	plansource = (CachedPlanSource *) linitial(plan->plancache_list);
 
-	/* Need _SPI_begin_call in case replanning invokes SPI-using functions */
-	SPI_result = _SPI_begin_call(false);
-	if (SPI_result < 0)
-		return false;
-
-	if (plan->saved)
-	{
-		/* Make sure the plan is up to date */
-		cplan = RevalidateCachedPlan(plansource, true);
-		ReleaseCachedPlan(cplan, true);
-	}
-
-	_SPI_end_call(false);
```

Comment:
Did you remove this code block to make it identical to 6X? Looks like `RevalidateCachedPlan` may modify `plansource`.

Reply:
The function of `SPI_is_cursor_plan()` where we have removed this code block is simple - to detect whether a cached plan is a cursor or not. Yes, the returned [result](https://github.com/arenadata/gpdb/blob/ea5ecfb070caa1b2c360fbda121fac2b06ad30f5/src/backend/executor/spi.c#L1435) below depends on `plansource->resultDesc`. We [rebuild](https://github.com/arenadata/gpdb/blob/ea5ecfb070caa1b2c360fbda121fac2b06ad30f5/src/backend/utils/cache/plancache.c#L588) `plansource` in `RevalidateCachedPlan()->RevalidateCachedPlanWithParams()` only [if the plan is](https://github.com/arenadata/gpdb/blob/ea5ecfb070caa1b2c360fbda121fac2b06ad30f5/src/backend/utils/cache/plancache.c#L499) `NULL`. But if it is `NULL` we [return](https://github.com/arenadata/gpdb/blob/ea5ecfb070caa1b2c360fbda121fac2b06ad30f5/src/backend/executor/spi.c#L1422) `false` in the code above and never get here.

Reply:
Looks like it is different `plan`s:

1. _SPI_plan in SPI_is_cursor_plan
1. CachedPlan inside RevalidateCachedPlanWithParams, member of plansource that member of mentioned _SPI_plan

Reply:
> Looks like it is different plans:

+1

It's not quite clear the intention to remove the fragment above. I think it have to be exposed in commit message.

Reply:
SPI_is_cursor_plan() is a check function that should not change the state of the plan underhood in some opaque way. Previously we revalidated a cached plan inside SPI_is_cursor_plan() if the one was saved. This revalidation was removed because it differs from 6X and seems to be redandunt:
1. RevalidateCachedPlan() would be called later one more time for the saved plan.
2. The only cached plan's attribute used between SPI_is_cursor_plan() and RevalidateCachedPlan() is a query_string, but it is not modified by a plan revalidation.

Reply:
@maksm90 extended the commit message